client/lib/login/index.js

```diff
@@ -97,7 +97,15 @@ export function getSignupUrl(
 
 	if ( isGutenboarding ) {
 		const langFragment = locale && locale !== 'en' ? `/${ locale }` : '';
-		signupUrl = '/new/plans?signup' + langFragment;
+		const defaultSignupUrl = '/new/plans?signup' + langFragment;
```

Comment:
Similarly, `langFragment` should go after `/plans` and before `?signup`, I'm not sure why it's appending it right now!